modules/integration-node/src/integration_tests.ts

```diff
@@ -34,30 +49,80 @@ export async function testVector ({ name, keysInfo, plainTextStream, cipherStrea
   }
 }
 
-export async function integrationTestVectors (vectorFile: string, tolerateFailures: number = 0, testName?: string) {
-  const tests = await getTestVectorIterator(vectorFile)
+// This is only viable for small streams, if we start get get larger streams, an stream equality should get written
+export async function testEncryptVector ({ name, keysInfo, encryptOp, plainTextData }: EncryptTestVectorInfo, decryptOracle: URL): Promise<TestVectorResults> {
+  try {
+    const cmm = encryptMaterialsManagerNode(keysInfo)
+    const { ciphertext } = await encrypt(cmm, plainTextData, encryptOp)
+
+    const decryptResponse = await got.post(decryptOracle, {
+      headers: {
+        'Content-Type': 'application/octet-stream',
+        'Accept': 'application/octet-stream'
+      },
+      body: ciphertext,
+      encoding: null
+    })
+    needs(decryptResponse.statusCode === 200, 'decrypt failure')
+    const { body } = decryptResponse
+    const result = plainTextData.equals(body)
+    return { result, name }
+  } catch (err) {
+    return { result: false, name, err }
+  }
+}
+
+export async function integrationDecryptTestVectors (vectorFile: string, tolerateFailures: number = 0, testName?: string) {
+  const tests = await getDecryptTestVectorIterator(vectorFile)
+  let failureCount = 0
+  for (const test of tests) {
+    if (testName) {
+      if (test.name !== testName) continue
+    }
+    const { result, name, err } = await testDecryptVector(test)
+    if (result) {
+      console.log({ name, result })
+    } else {
+      if (err && notSupportedDecryptMessages.includes(err.message)) {
```

Comment:
Do we want to enumerate the vectors that are not supported? Right now, if something returns not supported, but should be, it would slip by. That being said, this can go into a new issue, and doesn't need to be fixed here.

Reply:
or is that what "tolerateFailures" does?

Reply:
tolerateFailures is just a count of failures to tolerate.  CI should be 0.

Your comment is true.  The hope is to come up with a results manifest.  That could be compared and determine success.